src/workloads/query/TelemetryWithRedaction.yml

```diff
@@ -0,0 +1,193 @@
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/query"
+Description: |
+  This workload tests performance of $telemetry with and without redaction.
+
+KeyWords:
+- telemetry
+- redaction
+
+Clients:
+  Default:
+    QueryOptions:
+      maxPoolSize: 400
+
+GlobalDefaults:
+  Database: &Database test
+  Collection: &Collection Collection0
+  DocumentCount: &DocumentCount 1e6
+  MaxPhases: &MaxPhases 7
+  nop: &Nop {Nop: true}
+  dbname: &db test
+  binDataGen: &binDataGen {^BinData: {numBytes: 32}}
+
+ActorTemplates:
+- TemplateName: TelemetryAggregation
+  Config:
+    Name: {^Parameter: {Name: "Name", Default: "TelemetryAggregation"}}
+    Type: AdminCommand
+    Threads: 32
+    Phases:
+      OnlyActiveInPhases:
+        Active: [{^Parameter: {Name: "OnlyActiveInPhase", Default: 1024}}]
+        NopInPhasesUpTo: *MaxPhases
+        PhaseConfig:
+          Duration: 1 minute
+          SleepBefore: 5 seconds
+          Database: admin
+          Operations:
+          - OperationMetricsName: TelemetryWithRedactionMetrics
+            OperationName: AdminCommand
+            OperationCommand:
+              aggregate: 1
+              pipeline: [
+                      {
+                        $telemetry: {
+                          redactIdentifiers: {^Parameter: {Name: "redactIdentifiers", Default: false}},
+                          redactionKey: *binDataGen
+                        }
+                      }
+                    ]
+              cursor: {}
+
+Actors:
+- Name: ClearCollection
```

Comment:
I am now thinking - this will not clear the in-memory telemetry data, right? 

I don't know how we run things in evergreen in terms of a server restart or not, but it seems plausible that the $telemetry invocation here will have non-determenistic results based on what workloads ran before it. To ensure repeatable results, we'll probably want to clear it. I don't think there's any good way to do this other than setting the memory size limit to 0 and then back to what it was - which might be kinda challenging in this environment... I know there's a generic 'RunCommand' actor you could use to change the server parameter, but I don't think the YAML has a good way of remembering what the previous value of the parameter was - even if you ran getParameter to retrieve it. You could maybe skip the part about "back to what it was" since remembering the answer seems challenging to me on first thought. And just set it to be "1%" which we know is the default and seems reasonable. #performance-tooling-users might have some better ideas here.

Reply:
Can you add some of this context as a comment near the 'setParameter' command? I don't think it will be obvious to future readers, and I also want to make it clear that 250MB was kinda arbitrary.